src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -0,0 +1,263 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <gsl/gsl_integration.h>
+#include <memory>
+
+#include "Utilities/Gsl.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+/// Numerical integration algorithms
+namespace integration {
+
+namespace detail {
+
+// The GSL functions require the integrand to have this particular function
+// signature. In particular, any extra parameters to the functions must be
+// passed as a void* and re-interpreted appropriately.
+template <typename IntegrandType>
+double integrand(double x, void* params) {
```

Comment:
noexcept, can the pointer be `const`, can `x` be const?

Reply:
I've set x to be `const`. I get a compiler error if I set the pointer to be `const` though, because it does not match the type of `gsl_function`.

Reply:
Okay, did you also change the constness of the pointer in the `reinterpret_cast` below at the same time? I'm confused why the constness of the pointer (rather than the constness of the object being pointed to) would matter